src/main/java/org/tc/mail/MailNotificationSender.java

```diff
@@ -42,15 +49,15 @@ public void sendManagerNotification(Course course) {
         try {
             MimeMessageHelper helper = new MimeMessageHelper(message, true);
             helper.setSubject(COURSE_ANNOUNCEMENT_SUBJECT);
-            String[] emailsTo = {roleService.getDepartmentManager().getEmail(),
-                    roleService.getKnowledgeManager().getEmail()};
+            String[] emailsTo = {roleService.getDepartmentManager().get().getEmail(),
```

Comment:
As I can see usage of .get() shows that you used Optional but should you check that it exist? Otherwise it can throw NPE.